### ISSUE TITLE
Re-enable PHPCompatibility Sniffs

### ISSUE THREAD

**Issue by jrfnl** [Yoast/yoastcs#15]
I noticed that the [PHPCompatibility](https://github.com/wimg/PHPCompatibility) ruleset has been disabled as it - at the time - did not (yet) support PHPCS 2.x. This has been fixed for over a year now and since then a fairly complete suite of tests has been added to the PHPCompatibility library for PHP 7.

I'd like to suggest re-enabling it.

To that end, I've set up the basics for this in a feature branch, but this will need testing, preferably by someone a lot better versed in Composer than me ;-) _hint hint nudge nudge_

See: https://github.com/Yoast/yoastcs/tree/feature/re-enable-phpcompatibility

**Comment by Rarst** [comment]
As per https://github.com/wimg/PHPCompatibility/pull/234 it's sort of broken for Composer use at the moment.

I really don't like that plugin approach to move folder after install and it hadn't been _necessary_ with support for relative paths in PHP CS config.

**Comment by jrfnl** [comment]
> As per wimg/PHPCompatibility#234 it's sort of broken for Composer use at the moment.

There are a few issues open regarding that, but for most Composer installs - this issue does not apply. From all I've read about the issue, it's mostly a problem on Windows, so doesn't apply to travis runs.

> I really don't like that plugin approach to move folder after install and it hadn't been necessary with support for relative paths in PHP CS config.

Not sure what you mean exactly. Could you elaborate a bit ?

**Comment by Rarst** [comment]
It's not a problem on Windows, it's quite clearly a problem with current plugin setup. Its `autoload` directive in `composer.json` is effectively non-functional.

The way it _does_ work is using that additional plugin to physically move the standard inside PHP CS directory. I would say that is highly not in line with normal Composer mechanics and isn't necessary with current PHP CS versions. As you can see in our `composer.json` we just run configuration command which tells PHP CS where the standards are located in vendor.

**Comment by Rarst** [comment]
TL;DR its Composer setup is part broken, part unorthodox and I am not too keen to drag that into YoastCS and deal with it. :)

**Comment by jrfnl** [comment]
I see what you mean. That is actually a discussion I'm trying to resolve in the PHPCompatibility repo already. I've proposed a potential solution which would be more in line for Composer (moving the Sniffs one directory down), but as it would break any PEAR installs, it is not considered an acceptable solution ATM.

If you have any input on how to solve this better - it would be very welcome.

You can read the discussion on the various solution directions being considered here: https://github.com/wimg/PHPCompatibility/pull/185

**Comment by Rarst** [comment]
Yes, I looked through that. Unfortunately I don't have an idea how to work around folder name mismatch, since this ties into how PHP CS processes standards (which I think was covered in discussion there).

**Comment by jrfnl** [comment]
OK, thanks for looking and yes, that was covered in the discussion.

Let's leave this issue open for now. I'm still hoping to find a workable solution for PHPCompatibility which will work & be acceptable for both Composer as well as PEAR installs some time over the next few weeks (months).

**Comment by Rarst** [comment]
Sure, if I have any bright ideas I will update with.

**Comment by jrfnl** [comment]
Appreciated.

**Comment by jrfnl** [comment]
Looks like there is someone who's on the right track to solve this: https://github.com/wimg/PHPCompatibility/pull/235 :smile_cat: 

If you have a chance, would love to hear your opinion on the proposed solution. It is much closer to what you would expect with a Composer install. It just moves the relevant files to a subdirectory (in `vendor/wimg/PHPCompatibility`) so the installed path command will work.

**Comment by Rarst** [comment]
Does seem like a total hack. :) But _relatively_ it seems to be an improvement containing changes _within_ package rather than making them cross-package (if I follow that right).

Honestly I don't hope for a robust solution there without normalizing directory structure. If the standard is expected to be called one thing and it's not called that in the project then _every_ approach to build will have to deal with it, each in a different way.

I sympathize with backwards compatibility challenge there, but my concern for YoastCS is to keep _our_ mess on the low. :)

**Comment by jrfnl** [comment]
This is on track to be fixed in the next PHPCompatibility release 7.1.5 - see https://github.com/wimg/PHPCompatibility/pull/446

**Comment by jrfnl** [comment]
The above mentioned PR has been merged and will soon be released as PHPCompatibility 8.0.0.
